src/posts/2022-11-03-constellation/index.md

```diff
@@ -84,59 +84,67 @@ We define the following constraints for our strict mode:
 
 While the IPv4 limitation can be easily solved later, the PodCIDR must be static and known before deploying Cilium. Support for a dynamic PodCIDR is a complex feature to implement, because of the same eventual consistency problems we are trying to solve in the first place. Since we support Cilium’s tunneling and direct routing capabilities, the strict mode must also work with both. Note that the test we adapted already covers both scenarios.
 
-> A strict mode for AWS and Azure with direct routing poses some challenges. Since Pod IPs are allocated out of the same subnet as Node IPs, we cannot filter by subnet. Moreover, we cannot rely on the IPCache to add something like an allow-node rule to the filter. The Node could have been terminated and its IP re-used as a Pod IP while the IP is still associated with the Node in the IPCache. For that reason, we’re using vxlan.
- 
+### Direct routing
+
+As stated earlier, we only use direct routing on GCP and rely on vxlan for AWS and Azure. This is because GCP is the only of the three which implements IPAM in a way in which the PodCIDR is disjoint from the NodeCIDR.
+
+Implementing strict mode support for AWS and Azure with direct routing poses some challenges. Since Pod IPs are allocated out of the same subnet as Node IPs, we cannot filter by subnet. As this would also disrupt Node-to-Node communication. To regain Node-to-Node communication we can combine our CIDR-based filter with the IPCache. If a packet matches the filter we can lookup the identity of the destination in the IPCache. If the destination is a Node, we allow packet to leave the Node.
```

Comment:
small note here: It is possible to use a secondary VPC CIDR for pods on EKS and have Cilium use a subnet tag filter to only use subnets from the "pod VPC CIDR" for pods. But most users do have a setup where node and pod IPs are allocated from the same subnet.

But even then, you could have non-pod workloads in that subnet, so strict mode with direct routing might not be what most users want.

Reply:
Ah thanks for the info! I recently found that it is possible to separate Pods and Node CIDR on Azure but I haven't tried it out yet. Good to know that AWS also supports that. 
Should we rephrase the second sentence as `Most of the time, when using direct routing Pod IPs are allocated out of the same subnet as Node IPs we cannot filter by subnet.` ?

Reply:
Yeah, I think a rephrasing like this works well!